### DISCUSSION TITLE
Carla without gpu on docker ?

### DISCUSSION THREAD

**Discussion by NaveedBhuiyan** [carla-simulator/carla#6055] [Q&A]
I am trying to create a docker file for carla which will only run on the cpu with the no rendering mode. Problem is that in order to build the alreaedy existing docker image we need to install the nvidia drivers, which means we need gpu access. However I was hoping to create this docker file without any need to have any nvidia access for purely no rendering purposes. any ideas on whether this is possible and if so then how it is possible ??

**Comment by doganulus** [comment]
Check our repo: https://github.com/boundrivesim/starter-carla-0913

Dockerfile is under the server directory. You need to run with `-nullrhi` flag to disable the rendering completely. Then Carla engine can run on older computers without GPUs.

**Reply by NaveedBhuiyan** [reply]
hey,
Thanks for the reply. I followed the instruction in your repository but I keep getting this error when I start docker compose :
'error: XDG_RUNTIME_DIR not set in the environment'

Any ideas on how to fix this ?

**Reply by NaveedBhuiyan** [reply]
also...running the run.py script throws this error when the server is already running:
"ERROR:root:time-out of 2000ms while waiting for the simulator, make sure the simulator is ready and connected to 127.0.0.1:2000"

**Reply by doganulus** [reply]
The Carla Engine alone takes 4GB RAM and the start-up may be slow on low-end systems. 

Otherwise I can run it as described in the repo.

**Reply by NaveedBhuiyan** [reply]
I was able to make it work by increasing the world timeout from 2000ms  to 4000ms. One interesting thing I noticed was that the docker-compose.yml file does not actually build the Dockerfile in the server folder, but rather pulls an image from your repository. Am I correct here or am i missing something here? was there any reason for doing that ?
